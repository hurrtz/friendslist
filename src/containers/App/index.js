import React, { PureComponent } from 'react';
import { combineReducers, createStore } from 'redux';
import { Provider } from 'react-redux';

import FriendListApp from '../FriendListApp';
import * as reducers from '../../reducers';

const reducer = combineReducers(reducers);
const store = createStore(
  reducer,
  window.__REDUX_DEVTOOLS_EXTENSION__ && window.__REDUX_DEVTOOLS_EXTENSION__(),
);

class App extends PureComponent {
  render() {
    return (
      <div>
        <Provider store={store}>
          <FriendListApp />
        </Provider>
      </div>
    );
  }
}

export default App;
