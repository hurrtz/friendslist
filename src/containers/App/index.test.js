import React from 'react';
import Enzyme, { mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-15';
import renderer from 'react-test-renderer';

import App from './';

Enzyme.configure({ adapter: new Adapter() });

function setup() {
  const enzymeWrapper = mount(<App />);

  return { enzymeWrapper };
}

describe('App', () => {
  it('should render self and subcomponents', () => {
    const { enzymeWrapper, props } = setup();

    expect(enzymeWrapper.find('FriendListApp').exists()).toBe(true);
  });

  it('should match the snapshot', () => {
    const tree = renderer.create(<App />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
