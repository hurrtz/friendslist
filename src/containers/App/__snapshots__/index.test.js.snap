exports[`App should match the snapshot 1`] = `
<div>
  <div
    className={undefined}>
    <h1>
      The FriendList
    </h1>
    <input
      autoFocus="true"
      className="form-control"
      onChange={[Function]}
      onKeyDown={[Function]}
      placeholder="Type the name of a friend"
      type="text"
      value="" />
    <ul
      className={undefined}>
      <li
        className={undefined}
        onClick={[Function]}>
        <div
          className={undefined}>
          <div
            className={undefined}>
            <div>
              <span>
                Theodore Roosevelt
              </span>
            </div>
            <div>
              <small>
                xx friends in common
              </small>
            </div>
          </div>
          <div
            className={undefined}>
            <button
              className="btn btn-default"
              onClick={[Function]}>
              <i
                className="fa fa-star" />
            </button>
            <button
              className="btn btn-default"
              onClick={[Function]}>
              <i
                className="fa fa-trash" />
            </button>
          </div>
        </div>
        <hr
          className="undefined" />
        <div
          className="undefined">
          <div
            className="">
            <small>
              Sex:
            </small>
            <button
              className="btn btn-default undefined"
              onClick={[Function]}>
              <i
                className="fa fa-mars" />
            </button>
            <button
              className="btn btn-default"
              onClick={[Function]}>
              <i
                className="fa fa-venus" />
            </button>
          </div>
        </div>
      </li>
      <li
        className={undefined}
        onClick={[Function]}>
        <div
          className={undefined}>
          <div
            className={undefined}>
            <div>
              <span>
                Abraham Lincoln
              </span>
            </div>
            <div>
              <small>
                xx friends in common
              </small>
            </div>
          </div>
          <div
            className={undefined}>
            <button
              className="btn btn-default"
              onClick={[Function]}>
              <i
                className="fa fa-star-o" />
            </button>
            <button
              className="btn btn-default"
              onClick={[Function]}>
              <i
                className="fa fa-trash" />
            </button>
          </div>
        </div>
        <hr
          className="undefined" />
        <div
          className="undefined">
          <div
            className="">
            <small>
              Sex:
            </small>
            <button
              className="btn btn-default undefined"
              onClick={[Function]}>
              <i
                className="fa fa-mars" />
            </button>
            <button
              className="btn btn-default"
              onClick={[Function]}>
              <i
                className="fa fa-venus" />
            </button>
          </div>
        </div>
      </li>
    </ul>
    <div
      className={undefined}>
      <div />
      <button
        className="btn bnt-default"
        onClick={[Function]}
        type="button">
        Next
      </button>
    </div>
  </div>
</div>
`;
